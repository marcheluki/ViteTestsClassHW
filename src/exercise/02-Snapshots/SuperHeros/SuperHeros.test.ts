/* Exercise 2: Test using snapshots */

/* Mock the function using jest.fn().
Write three tests inside a describe block. You should import the superHeros[] and getFlyingSuperHeros function.

1. First Test: 
Test should return an empty array if no superheros have the 'Fly' power
2. Second Test:
Test should return an array of superheros that have the 'Fly' power
3. Third Test:
Test should match the snapshot of flying superheros.
The snapshot file should contain the expected output of the test.
The snapshot should be saved in a __snapshots__ directory next to the test file.
The snapshot file should be named SuperHeros.test.ts.snap.
*/

import { getFlyingSuperHeros } from "./getFlyingSuperHeros";
import { superHeros } from "./superHeros";

describe("SuperHeros tests", () => {
  it("should return an empty array if no superheros have the 'Fly' power", () => {
    const result = getFlyingSuperHeros([]);
    expect(result).toEqual([]);
  });

  it("should return an array of superheros that have the 'Fly' power", () => {
    const result = getFlyingSuperHeros(superHeros);
    // Building the expected array using the same logic that getFlyingSuperHeros uses
    const expected = superHeros.filter(hero => hero.power.includes("Fly"));
    expect(result).toEqual(expected);
  });

  it("should match the snapshot of flying superheros", () => {
    const result = getFlyingSuperHeros(superHeros);
    expect(result).toMatchSnapshot();
  });
});

