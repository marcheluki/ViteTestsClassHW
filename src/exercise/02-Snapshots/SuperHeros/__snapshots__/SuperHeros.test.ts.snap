// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuperHeros tests should match the snapshot of flying superheros 1`] = `
[
  {
    "name": "Superman",
    "power": [
      "Fly",
      "Super Strength",
    ],
  },
  {
    "name": "IronMan",
    "power": [
      "Intelligence",
      "Technology",
      "Fly",
      "Billionaire",
    ],
  },
  {
    "name": "GreenLantern",
    "power": [
      "Energy Manipulation",
      "Fly",
    ],
  },
]
`;
